src/components/lib/cos_kernel_api.c

```diff
@@ -276,18 +283,14 @@ __capid_bump_alloc(struct cos_compinfo *ci, cap_t cap)
 /**************** [User Virtual Memory Allocation Functions] ****************/
 
 static vaddr_t
-__page_bump_valloc(struct cos_compinfo *ci)
+__bump_mem_expand_range(struct cos_compinfo *ci, pgtblcap_t cipgtbl, vaddr_t mem_ptr, unsigned long mem_sz)
 {
-	vaddr_t heap_vaddr;
+	vaddr_t addr;
 	struct cos_compinfo *meta = __compinfo_metacap(ci);
 
-	printd("__page_bump_alloc\n");
-
 	assert(meta == __compinfo_metacap(meta)); /* prevent unbounded structures */
-	heap_vaddr = ci->vas_frontier;
 
-	/* Do we need to allocate a PTE? */
-	if (heap_vaddr == ci->vasrange_frontier) {
+	for (addr = mem_ptr; addr < round_up_to_pgd_page(mem_ptr + mem_sz); addr += PGD_RANGE) {
```

Comment:
`for (addr = mem_ptr ; addr < round_up_to_pgd_page(mem_ptr + mem_sz) ; addr += PGD_RANGE) {`

This needs the emphasis around the 3 parts.  Style guide.